src/bvh/iter.rs

```diff
@@ -107,6 +107,7 @@ impl<'bvh, 'shape, T: BHValue, const D: usize, Shape: Bounded<T, D>>
     }
 }
 
+#[allow(clippy::needless_lifetimes)]
 impl<'bvh, 'shape, T: BHValue, const D: usize, Shape: Bounded<T, D>> Iterator
     for BvhTraverseIterator<'bvh, 'shape, T, D, Shape>
```

Comment:
Can't we do this instead?
```suggestion
impl<'shape, T: BHValue, const D: usize, Shape: Bounded<T, D>> Iterator
    for BvhTraverseIterator<'_, 'shape, T, D, Shape>
```

Reply:
Yes, that works too! I originally chose `#[allow(clippy::needless_lifetimes)]` because I didn't know whether the needless lifetime with a special name "bvh" was unintentional or intentional (for consistent and self-documentation).

Reply:
I _do_ like the consistency of it but I think `#[allow()]` things are really a last resort. I think this case is really 50/50 but let's go with clippy's suggestion.